src/utils/MasterFisher.ts

```diff
@@ -0,0 +1,104 @@
+import { Fishable, Type, Metadata } from './Fishable';
+import { FSHTank } from '../import';
+import { FHIRDefinitions } from '../fhirdefs';
+import { Package } from '../export';
+import { logger } from './FSHLogger';
+
+/**
+ * The MasterFisher can fish from the tank, the FHIR definitions, and the package that is currently
+ * being exported. When the MasterFisher fishes, it fishes in the package first, then the tank, and
+ * then the FHIRDefinitions.  This essentially prefers local definitions first (when there are naming
+ * clashes).
+ *
+ * The MasterFisher also uses its knowledge of these Fishable locations to do some necessary magic.
+ * For instance, the FSHTank has no knowledge of FHIRDefinitions, so it cannot provide the correct
+ * sdType for its metadata results.  When the MasterFisher detects this has happened, it uses the
+ * other Fishable locations to determine the proper sdType (even for an item that currently exists
+ * only in the tank).
+ */
+export class MasterFisher implements Fishable {
+  constructor(public tank?: FSHTank, public fhir?: FHIRDefinitions, public pkg?: Package) {}
+
+  /**
+   * Searches for the FHIR JSON by name/id/url.  It will first search through the local package
+   * (which contains FHIR artifacts exported so far), then through the external FHIR definitions.
+   * @param {string} item - the item name/id/url to fish for
+   * @param types - the allowable types to fish for
+   */
+  fishForFHIR(item: string, ...types: Type[]): any | undefined {
+    // Resolve the alias if necessary
+    item = this.tank?.resolveAlias(item) ?? item;
+
+    // First check for it in the package
+    let result = this.pkg?.fishForFHIR(item, ...types);
+    if (result == null) {
+      // If it is in the tank, return undefined. We don't want to return the external FHIR
+      // definition, even if it exists -- because it won't match what is in the tank.  This
+      // ensures consistency between the outputs of fishForFHIR and fishForMetadata.
```

Comment:
Yes, that is exactly right!  The second bullet ensures we avoid the case where there is something in the tank with a name or id that matches something in FHIR.  For example, in mCODE, there is a `laterality` extension that mCODE defines.  As it happens, FHIR core also has a `laterality` extension.  If we fish for `laterality` but we haven't exported it yet, we _don't_ want to return FHIR core `laterality` because that's not the right `laterality`.